internal/jimm/jimm.go

```diff
@@ -730,25 +731,45 @@ func fillMigrationTarget(db *db.Database, credStore credentials.CredentialStore,
 }
 
 // InitiateInternalMigration initiates a model migration between two controllers within JIMM.
-func (j *JIMM) InitiateInternalMigration(ctx context.Context, user *openfga.User, modelTag names.ModelTag, targetController string) (jujuparams.InitiateMigrationResult, error) {
+func (j *JIMM) InitiateInternalMigration(ctx context.Context, user *openfga.User, modelNameOrUUID string, targetController string) (jujuparams.InitiateMigrationResult, error) {
 	const op = errors.Op("jimm.InitiateInternalMigration")
 
 	migrationTarget, _, err := fillMigrationTarget(j.Database, j.CredentialStore, targetController)
 	if err != nil {
 		return jujuparams.InitiateMigrationResult{}, errors.E(op, err)
 	}
-	// Check that the model exists
-	model := dbmodel.Model{
-		UUID: sql.NullString{
-			String: modelTag.Id(),
+
+	model := dbmodel.Model{}
+	// Check if the user is providing a model UUID or name
+	_, err = uuid.Parse(modelNameOrUUID)
+	if err != nil {
+		s := strings.Split(modelNameOrUUID, "/")
+		if len(s) != 2 {
+			return jujuparams.InitiateMigrationResult{}, errors.E(op, "missing model name")
```

Comment:
this is more "invalid target" right? because we don't know if the model or the user name is wrong

Reply:
Yeah true, I'll update it